app/scripts/lib/app-start.js

```diff
@@ -257,21 +257,25 @@ define(function (require, exports, module) {
         let relier;
         const context = this._getContext();
 
-        if (this._isServiceSync()) {
-          // Use the SyncRelier for sync verification so that
-          // the service name is translated correctly.
-          relier = new SyncRelier({ context }, {
+        // The check for whether the user is in an OAuth flow
+        // is done first to ensure OAuth reliers that specify
+        // service=sync in addition to client_id still get the
```

Comment:
How about this for text, trying to explain the rationale:

> The order of the checks is important. The OAuth check 
is more strict than the Sync check, and if the Sync 
check is done first, a bad acting OAuth relier could 
cause us to present the Sync UI to the user.

The obvious question becomes:

> Why can't we make the Sync check as strict as the OAuth check?

Because Sync verification in a 2nd browser only has `code`, `uid`, and `service` to determine the user is in a Sync flow.